### DISCUSSION TITLE
wagmi/cli: Allow etherscan plugin to generate hooks with overridable address param

### DISCUSSION THREAD

**Discussion by azevedo** [wevm/wagmi#2102] [Ideas]
When using the [etherscan plugin](https://wagmi.sh/cli/plugins/etherscan) we need to specify a contract address to fetch the ABI from.
This results in the generated hooks always using that same address, and even the `address` param being omitted from the hooks' `config` param, like so
```ts
export function useStrategy(
  config: Omit<UseContractConfig, "abi" | "address"> & { // <---------- address is omitted
    chainId?: keyof typeof strategyAddress
  } = {} as any
) {
  return useContract({
    abi: strategyABI,
    address: strategyAddress[5], // <---------- it passes the address defined in wagmi.config
    ...config,
  })
}
```
There are cases where we may have multiple contracts sharing the same ABI, in which case, it would be nice to either:
* allow the user to override the `address` config param
* not pass any address to `useContract` by default — I know that when [passing an inline ABI to the `contracts` property](https://wagmi.sh/cli/configuration/options#abi) instead of using etherscan, it works just like that, generating this:

```ts
export function useStrategy(
  config: Omit<UseContractConfig, "abi"> = {} as any
) {
  return useContract({ abi: strategyABI, ...config })
}
```

**Comment by tmm** [comment]
Good idea! Would accept a contribution for this. This is somewhat related as well: https://github.com/wagmi-dev/wagmi/discussions/1931 Seems like there are a few approaches for `address`:

1. `address` omitted from generated hooks/actions and defaulted to `etherscan.contracts[number].address` (current behavior)
2. `address` resolved to implementation from proxy, omitted from generated hooks/actions, and defaulted internally #1931
3. `address` used for fetching ABIs, but not used internally or omitted from generated hooks/actions
